### PR TITLE
Clear the key packages and app config on ENError = 2

### PR DESCRIPTION
**Pull request by ArturFriesen**
## Description
Resets the key packages and app config when ENError = 2 (badParameter) occurs.

## Link to Jira
[3297](https://jira.itc.sap.com/browse/EXPOSUREAPP-3297)

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/AppDelegate.swift [hunk 1]
```diff
@@ -109,11 +109,6 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 		// use a custom http client that uses/recognized caching mechanisms
 		let appFetchingClient = CachingHTTPClient(clientConfiguration: client.configuration)
 
-		// we currently use the store as common place for temporal persistency
-		guard let store = store as? AppConfigCaching else {
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/AppDelegate.swift:113]: 👍

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetectionExecutor.swift [hunk 2]
```diff
@@ -168,11 +168,26 @@ final class ExposureDetectionExecutor: ExposureDetectionDelegate {
...
+		// Clear the key packages and app config on ENError = 2 = .badParameter
+		// For more details, see: https://jira.itc.sap.com/browse/EXPOSUREAPP-3297
+		func clearCacheOnError2(error: Error) {
+			if let enError = error as? ENError, enError.code == .badParameter {
+				// Clear the key packages
+				downloadedPackagesStore.reset()
+				downloadedPackagesStore.open()
+
+				// Clear the app config
+				store.appConfig = nil
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetectionExecutor.swift:181]: Just to make sure, every change to the store gets instantly persisted to Disk?

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetectionExecutor.swift (hunk 2 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetectionExecutor.swift:181]: Its reseting the app config cache. Also added etag and date reset.

Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetectionExecutor.swift:181]: Its persisted here: https://github.com/corona-warn-app/cwa-app-ios/blob/3c8416d309651783b2c1dafff446661bcac14e9d/src/xcode/ENA/ENA/Source/Workers/SQLiteKeyValueStore.swift#L138

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetectionExecutor.swift [hunk 3]
```diff
@@ -168,11 +168,28 @@ final class ExposureDetectionExecutor: ExposureDetectionDelegate {
 			writtenPackages: WrittenPackages,
 			completion: @escaping (Result<ENExposureDetectionSummary, Error>) -> Void
 	) -> Progress {
+
+		// Clear the key packages and app config on ENError = 2 = .badParameter
+		// For more details, see: https://jira.itc.sap.com/browse/EXPOSUREAPP-3297
+		func clearCacheOnError2(error: Error) {
```
Comment (KaiTeuber) [inline src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetectionExecutor.swift:174]: why call this func ..OnError2 and explain the error code in the comment?
I think we should use a better name here, like
clearCacheOnErrorBadParameter

**Review by KaiTeuber — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetectionExecutor.swift (hunk 3 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetectionExecutor.swift]: :) Renamed!

**Pull request merged**